core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -82,19 +82,57 @@ class GroupMetadataManager(brokerId: Int,
 
   this.logIdent = s"[GroupMetadataManager brokerId=$brokerId] "
 
-  newGauge("NumOffsets",
+  private def recreateGauge[T](name: String, gauge: Gauge[T]): Gauge[T] = {
```

Comment:
Not sure why we want to remove-then-recreate here? Isn't this a one-time call for the life time?